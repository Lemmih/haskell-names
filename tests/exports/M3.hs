module M3 where

data A = B { c :: A }

newtype G = H A

type L = G
