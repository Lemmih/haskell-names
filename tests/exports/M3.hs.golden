Symbols (
  fromList
    [ SymSelector
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "c"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "A"
                          }
        }
    , SymConstructor
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "B"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "A"
                          }
        }
    , SymConstructor
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "H"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "G"
                          }
        }
    ] ) (
  fromList
    [ SymType
        { st_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "L"
                          }
        , st_fixity = Nothing
        }
    , SymData
        { st_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "A"
                          }
        , st_fixity = Nothing
        }
    , SymNewType
        { st_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "M3"
                                          "G"
                          }
        , st_fixity = Nothing
        }
    ] )